### PR TITLE
#7201: Add support for unary ne

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Add support for Unary Not equal ( TTNN migration included)
All post-commit tests : [Link](https://github.com/tenstorrent-metal/tt-metal/actions/runs/8614084124) - PASSED
Test Results: [Link](https://github.com/tenstorrent-metal/tt-metal/pull/7234#issuecomment-2056915887)

TTNN changes removed : [Link](https://github.com/tenstorrent/tt-metal/pull/7234#issuecomment-2063511903)

### CODE REVIEW

**Review by eyonland — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
Test results : 
File : `tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_eltwise_unary.py`
Screenshot 2024-04-15 at 7 11 13 PM

File : `tests/ttnn/unit_tests/operations/test_relational.py`
Screenshot 2024-04-15 at 7 14 02 PM

File: tt_metal/include/compute_kernel_api.h [hunk 1]
```diff
@@ -533,6 +533,31 @@ ALWI void heaviside_tile_init() {
...
+ * in DST register at index tile_index. The value is provided as const param0 The DST register buffer must be in
+ * acquired state via *acquire_dst* call. This call is blocking and is only
+ * available on the compute engine.
+ *
+ * Return value: None
+ *
+ * | Argument        | Description                                                                | Type     | Valid Range                                           | Required |
+ * |-----------------|----------------------------------------------------------------------------|----------|-------------------------------------------------------|----------|
+ * | idst            | The index of the tile in DST register buffer to perform the computation on | uint32_t | Must be less than the size of the DST register buffer | True     |
+ * | param0          | The value the output is if the input is greater than 0                     | uint32_t |                                                       | True     |
```
Comment (tt-aho) [inline tt_metal/include/compute_kernel_api.h:548]: Fix description of param0

File: tt_metal/include/compute_kernel_api.h [hunk 2]
```diff
@@ -533,6 +533,31 @@ ALWI void heaviside_tile_init() {
     MATH(( llk_math_eltwise_unary_sfpu_heaviside_init<APPROX>() ));
 }
 
+//unary ne : if x !=value --> 1.0, else 0.0
+/**
+ * Performs element-wise computation of:  result = 0 if x!=value , where x is each element of a tile
```
Comment (tt-aho) [inline tt_metal/include/compute_kernel_api.h:538]: First comment says output is 1 if x != value,
next comment says result = 0 if x!=value

**Review by tt-aho — CHANGES_REQUESTED** [review]

**Review by rtawfik01 — APPROVED** [review]

**Review by arakhmati — CHANGES_REQUESTED** [review]
What is the difference between the new op and `tt_lib.tensor.ne`?

**Comment by VirdhatchaniKN** [comment]
> What is the difference between the new op and `tt_lib.tensor.ne`?

`ne` takes in two tensors as input and `unary_ne` takes in a tensor and scalar as input.

**Review by tt-aho — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
TTNN Changes removed

- All post-commit tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/8734935584) - PASSED
- Fast Dispatch unit tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/8734936541) - PASSED
- Slow Dispatch unit tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/8734937638) - PASSED

**Review by arakhmati — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
- All post-commit tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/8812832050) - PASSED
- Fast Dispatch unit tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/8812833621) - PASSED
- Slow Dispatch unit tests - [Link](https://github.com/tenstorrent/tt-metal/actions/runs/8812835205) - PASSED

**Pull request merged**